File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/RookEnvironmentMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.environment.RookEnvironment
import com.rookmotion.rook_sdk_health_connect.data.proto.RookEnvironmentProto

fun RookEnvironmentProto.toRookEnvironment(): RookEnvironment {
    return if (this == RookEnvironmentProto.SANDBOX) {
        RookEnvironment.SANDBOX
    } else {
        RookEnvironment.PRODUCTION
    }
}

File: packages/rook_sdk_health_connect/lib/src/data/mapper/rook_environment_mappers.dart
import 'package:rook_sdk_core/rook_sdk_core.dart';
import 'package:rook_sdk_health_connect/src/data/proto/protos.pbenum.dart';

extension RookEnvironmentMappers on RookEnvironment {
  RookEnvironmentProto toProto() {
    return switch (this) {
      RookEnvironment.sandbox => RookEnvironmentProto.SANDBOX,
      _ => RookEnvironmentProto.PRODUCTION,
    };
  }
}
